pbft/pbft.cpp

```diff
@@ -1101,6 +1101,8 @@ pbft::handle_database_message(const bzn_envelope& msg, std::shared_ptr<bzn::sess
 {
     // TODO: timestamp should be set by the client. setting it here breaks the signature (which is correct).
 
+    std::lock_guard<std::mutex> lock(this->pbft_lock);
```

Comment:
Move this inside the if block. This removes the gains we made with switching to a shared lock when qreads are executed.

Reply:
Done